src/loci/formats/in/ZarrReader.java

```diff
@@ -415,16 +536,29 @@ else if (image instanceof double[]) {
 
   @Override
   public void setSeries(int no) {
+    setSeries(no, false);
+  }
+  
+  public void setSeries(int no, boolean openZarr) {
     super.setSeries(no);
-    openZarr();
+    if (openZarr) {
+      openZarr();
+    }
   }
 
   @Override
   public void setResolution(int no) {
+    setResolution(no, false);
+  }
+  
+  public void setResolution(int no, boolean openZarr) {
```

Comment:
What are the sensible usages of these new public method?

Reply:
Yeah thats probably worth discussing, depending on how the reader is realistically going to be used. An alternative would be to remove the new methods and move the condition statements from [here](71931eb54d3ee0b5a673044b8cdcdb8fa1bee03c583e3fc916e1b6b47a7dd7bbR287) into the original method. This would be a cleaner solution for most users, though it would add a small performance hit and reduce the flexibility. 

The initial functionality was that everytime a call was made to setSeries or setResolution, that a call would also be made to JZarr to set the relevant .zarray. From the initial profiling it was apparent that a huge number of these calls were being made and that it may not always be required to make the jZarr call. The most performant option was to not make the JZarr call by default, but on some occasions it is required (such as to initially fetch the shapes), and as such a new method was needed.

Setting the new methods to private would likely suffice for most use cases also. In that case we would have to force the Jzarr open call into methods such as getOptimalTileSize. As in that case we would need to ensure the chunk sizes are being read from the correct zarray.